### DISCUSSION TITLE
Is there support for overlayfs without fuse-overlayfs for rootless?

### DISCUSSION THREAD

**Discussion by eriksjolund** [containers/podman#9772] [Q&A] [answered]
According to
https://kernelnewbies.org/Linux_5.11#Unprivileged_Overlayfs_mounts
Linux kernel 5.11 "... _allows non-privileged users to make use of Overlayfs_"

In the list _Shortcomings of Rootless Podman_
https://github.com/containers/podman/blob/master/rootless.md
it's written:
"_Can not use overlayfs driver, but does support fuse-overlayfs_"

Is there already support for _overlayfs_ (without using _fuse-overlayfs_) for rootless Podman?
(The list of shortcomings indicates that the answer is no but maybe things are changing?)

When I ran `sudo dnf update`  today on my Fedora 33 computer the kernel was updated to version 5.11.7.

```
$ uname -r
5.11.7-200.fc33.x86_64
$ podman --version
podman version 3.0.1
$
```

**Comment by rhatdan** [comment, marked as answer]
@giuseppe began work on this, but I think there might be still a bug in overlay not supporting context (SELinux) mounts in rootless mode.

https://github.com/containers/storage/pull/845

This will be vendored into podman 3.1, but I don't know if it will fully work.

@giuseppe what is the status on this?

**Reply by giuseppe** [reply]
yes, there is a bug in overlay that prevents us to use it from rootless.

We need to wait for Linux 5.12.

**Reply by giuseppe** [reply]
The fix is here: https://www.spinics.net/lists/selinux/msg35984.html
